src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -482,12 +483,23 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                     }
                 }
 
-                if (!((CurrentToken?.Kind is HttpTokenKind.Word or HttpTokenKind.Punctuation) || (IsAtStartOfEmbeddedExpression())))
+                if(CurrentToken?.Kind is HttpTokenKind.NewLine)
                 {
                     break;
                 }
 
-                if (IsAtStartOfEmbeddedExpression())
+                if ((CurrentToken?.Kind is HttpTokenKind.Whitespace))
+                {
+                    var nextSignificantToken = GetNextSignificantToken()?.Text.ToLowerInvariant();
+                    if ((nextSignificantToken is not "http" or "https"))
+                    {
+                        ConsumeCurrentTokenInto(node);
```

Comment:
Does the encoding of whitespace matter? If yes, do we have tests to make sure whitespace is encoded (`%20`) correctly in the resulting `System.Uri` object returned when we bind the url?